packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -992,7 +976,13 @@ class RenderClipRRect extends RenderProxyBox {
   void paint(PaintingContext context, Offset offset) {
     if (child != null) {
       Rect rect = Point.origin & size;
-      RRect rrect = new RRect.fromRectXY(rect, xRadius, yRadius);
+      RRect rrect = new RRect.fromRectAndCorners(
+        rect,
+        topLeft: borderRadius.topLeft,
+        topRight: borderRadius.topRight,
+        bottomRight: borderRadius.bottomRight,
+        bottomLeft: borderRadius.bottomLeft
+      );
```

Comment:
BorderRadius should have a toRRect() method that takes a Rect and returns an RRect so that you don't have to keep having this block of code everywhere.